### ISSUE TITLE
Function parameters ending with ${int} in source aren't escaped, and shadow higher scoped generated variables

### ISSUE THREAD

**Issue by bloodyowl** [rollup/rollup#2683]
Context:
[BuckleScript](https://github.com/BuckleScript/bucklescript) generates variables like `name$1`.

- Rollup Version: v1.1.2
- Operating System (or Browser): All
- Node Version: v10.14.2

### Input

```js
// main.js
import {x} from "./maths.js";

console.log(x(value));
```

```js
// maths.js
var value = "1";

export function x(value$1) { return value$1 + value } 
```

### Expected Behavior

```js
var value$1 = "1";

function x(value$1$1) { return value$1$1 + value$1 }

console.log(x(value));
```

or

```js
var value$1 = "1";

function x(value_$_1) { return value_$_1 + value$1 }

console.log(x(value));
```

### Actual Behavior

```js
var value$1 = "1";

function x(value$1) { return value$1 + value$1 }

console.log(x(value));
```

https://rollupjs.org/repl?version=1.1.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QnglN0QlMjBmcm9tJTIwJTVDJTIyLiUyRm1hdGhzLmpzJTVDJTIyJTNCJTVDbiU1Q25jb25zb2xlLmxvZyh4KHZhbHVlKSklM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJtYXRocy5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJ2YXIlMjB2YWx1ZSUyMCUzRCUyMCU1QyUyMjElNUMlMjIlM0IlNUNuJTVDbmV4cG9ydCUyMGZ1bmN0aW9uJTIweCh2YWx1ZSUyNDEpJTIwJTdCJTIwcmV0dXJuJTIwdmFsdWUlMjQxJTIwJTJCJTIwdmFsdWUlMjAlN0QlMjAlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXNtJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

**Comment by lukastaegert** [comment]
Thanks, the deconflicting logic rewrite I am currently working on should be able to handle this, will definitely include your test case!

**Comment by lukastaegert** [comment]
Fix a #2689
